staging/src/k8s.io/apiserver/pkg/server/filters/timeout_test.go

```diff
@@ -360,6 +360,59 @@ func TestTimeoutWithLogging(t *testing.T) {
 	res.Body.Close()
 }
 
+func TestClientSideCancel(t *testing.T) {
+	origReallyCrash := runtime.ReallyCrash
+	runtime.ReallyCrash = false
+	defer func() {
+		runtime.ReallyCrash = origReallyCrash
+	}()
+
+	postTimeoutCh := make(chan struct{})
+	postTimeoutFn := func() {
+		close(postTimeoutCh)
+	}
+
+	ts := httptest.NewServer(
+		WithHTTPLogging(
+			WithTimeout(
+				http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+					time.Sleep(30 * time.Second)
+					w.Write([]byte("ok"))
+				}),
+				func(req *http.Request) (*http.Request, bool, func(), *apierrors.StatusError) {
+					return req, false, postTimeoutFn, apierrors.NewTimeoutError("timed out", 0)
+				},
+			),
+		),
+	)
+	defer ts.Close()
+
+	client := &http.Client{}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPatch, ts.URL, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Cancel the request after 1 second.
+	go func() {
+		time.Sleep(time.Second)
+		cancel()
+	}()
+	if _, err := client.Do(req); err == nil {
```

Comment:
should you not try to assert the error message?

Reply:
This is always a context canceled error, regardless of the server behaviour (canceling the client side context causes this to return immediately, ignoring the HTTP response). I asserted the error type anyway.